File: src/Nebula/Modules/Sales/Models/CreditNote.cs
namespace Nebula.Modules.Sales.Models;

public class CreditNote : BaseSale
{
    public string InvoiceSaleId { get; set; } = string.Empty;
    public string CodMotivo { get; set; } = string.Empty;
    public string DesMotivo { get; set; } = string.Empty;
    public string TipDocAfectado { get; set; } = string.Empty;
    public string NumDocfectado { get; set; } = string.Empty;
}

File: src/angular/src/app/sales/interfaces/credit-note.ts
import {Company} from "app/account/interfaces";
import {BaseSale, SaleDetail} from "./invoice-sale";

export class CreditNote extends BaseSale {
  invoiceSaleId: string = "";
  codMotivo: string = "";
  desMotivo: string = "";
  tipDocAfectado: string = "";
  numDocfectado: string = "";
}

export class CreditNoteDetail extends SaleDetail {
  creditNoteId: string = "";
}

export class PrintCreditNoteDto {
  company: Company = new Company();
  creditNote: CreditNote = new CreditNote();
  creditNoteDetails: CreditNoteDetail[] = [];
}
